main/uart_nic.c

```diff
@@ -65,50 +61,57 @@ static const uint16_t INACTIVE_BEACON_SECONDS = 3600 * 18;
 // pings to the AP?
 static const uint32_t INACTIVE_PACKET_SECONDS = 5;
 
-// intron
-// 0 as uint8_t
-// fw version as uint16_t
-// hw addr data as uint8_t[6]
-#define MSG_DEVINFO 0
-
-// intron
-// 1 as uint8_t
-// link up as bool (uint8_t)
-#define MSG_LINK 1
-
-// intron
-// 2 as uint8_t
-#define MSG_GET_LINK 2
-
-// intron
-// 2 as uint8_t
-// ssid size as uint8_t
-// ssid bytes
-// pass size as uint8_t
-// pass bytes
-#define MSG_CLIENTCONFIG 3
-
-// intron
-// 3 as uint8_t
-// LEN as uint32_t
-// DATA
-#define MSG_PACKET 4
-
-// intron
-// 5 as uint8_t
-// new intron as uint8_t[8]
-#define MSG_INTRON 5
+// Note: Values 1..5 are reserved and must not be used.
+#define MSG_DEVINFO_V2 0
+#define MSG_CLIENTCONFIG_V2 6
+#define MSG_PACKET_V2 7
+
+struct header {
+    uint8_t type;
+    union {
+        uint8_t version; // when type == MSG_DEVINFO_V2
+        uint8_t unused;  // when type == MSG_CLIENTCONFIG_V2
+        uint8_t up;      // when type == MSG_PACKET_V2
+    };
+    uint8_t size[2];
+};
+
+struct uart0_tx_queue_item {
+    struct header header;
+    uint8_t *data;
+};
```

Comment:
I wonder if this is the right place to abuse the „Flexible array member“ feature of C 😈

(Not a request for a change, just noticing)

Reply:
It is not a flexible array member. `uart0_tx_queue_item` is just struct of size 8 (4 bytes header + 4 bytes pointer)